project/src/main/save-converter.gd

```diff
@@ -0,0 +1,98 @@
+class_name SaveConverter
+"""
+Provides backwards compatibility with older save formats.
+
+SaveConverter can update the 'version' tag, but any other version-specific updates must be defined externally. These
+version-specific updates can be incorporated via SaveConverter's 'add_method' method.
+"""
+
+"""
+An externally defined method which provides version-specific updates.
+"""
+class ConversionMethod:
+	var object: Object # The object containing the method
+	var method: String # The name of the method which performs the conversion
+	var old_version: String # The old save data version which the method converts from
+	var new_version: String # The new save data version which the method converts to
+
+# Externally defined methods which provide version-specific updates.
+# key: old save data version from which the method converts
+# value: a ConversionMethod corresponding to the method to call
+var conversion_methods := {}
+
+"""
+Adds a new externally defined method which provides version-specific updates.
+
+SaveConverter does not have logic for converting specific save data versions. This conversion logic must be defined on
+an external object and incorporated via this 'add_method' method.
+
+Parameters:
+	'object': The object containing the method
+	
+	'method': The name of the method which performs the conversion
```

Comment:
The comment should describe more details about the signature and contract of this method.

Reply:
Expanded comment, `This method should accept a SaveData object and return a modified SaveData object.`